src/components/AppModal.vue

```diff
@@ -20,14 +17,33 @@
  */
 export default {
   name: 'app-modal',
-  props: ['id'],
+  props: {
+    visible: Boolean,
+  },
+  watch: {
+    visible: {
+      handler(to) {
+        if (to) document.addEventListener('keyup', this.checkKey)
+        else document.removeEventListener('keyup', this.checkKey)
+      },
+      immediate: true,
+    },
+  },
+  destroyed() {
+    document.removeEventListener('keyup', this.checkKey)
+  },
+  methods: {
+    checkKey(e) {
+      if (e.keyCode === 27) this.$emit('close')
+    },
+  },
```

Comment:
Modal visibility will always be toggled by `v-show` on the `<app-modal />`. For that reason I would prefer to add/remove the key listener on mount/destroy instead of maintaining an internal 'visible' state.